src/bsp/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+set(LIB_NAME "bsp")
+set(LIB_ALIAS "SwarmUS::HiveMind::BSP")
+
+get_filename_component(BSP_CMAKE_DIR ${CMAKE_CURRENT_LIST_FILE} DIRECTORY)
+
+add_subdirectory(src)
+
```

Comment:
What do you mean by include all the files? We will add them as it goes. If you mean using GLOB, this is bad practice since your cmake will not be rebuilt when you add a new file.